specs-go/config.go

```diff
@@ -627,6 +627,19 @@ type WindowsCPUResources struct {
 	// cycles per 10,000 cycles. Set processor `maximum` to a percentage times
 	// 100.
 	Maximum *uint16 `json:"maximum,omitempty"`
+	// Set of CPUs to affinitize for this container.
+	AffinityCPUs []WindowsCPUGroupAffinity `json:"affinityCPUs,omitempty"`
+	// Specifies preferred set of numa node numbers to affinitize for this container.
+	AffinityPreferredNumaNodes []int64 `json:"affinityPreferredNumaNodes,omitempty"`
```

Comment:
What is actually done with this value?

Reply:
This would be used with https://learn.microsoft.com/en-us/windows/win32/api/processthreadsapi/nf-processthreadsapi-updateprocthreadattribute to set the Preffered numa node when using the memory manager.  

I have some of the details in https://github.com/kubernetes/enhancements/pull/4738/files#diff-69a2ffb8618949f1eddc42d0810c7afa17f3a82d8d8d7dbf07ba85495fd7137a

Reply:
I just commented here: https://github.com/kubernetes/enhancements/pull/4738/files#r1661653261

If we know preferred NUMA node via PROC_THREAD_ATTRIBUTE and affinity via job object are effectively the same thing (except one works at node granularity and the other at CPU granularity) do we need to offer both as options? Can't we just expose CPU affinity?

Reply:
@kevpar I don't see enough evidence to say PROC_THREAD_ATTRIBUTE  under covers is actually using cpu affinity mask in SetInformationJobObjectEx(). The KEP is going to be reworded a little. So I think it would be good to still have the numa node numbers passed down so hcsshim make the decision to set the masks appropriately. @jsturtevant does this sound ok?

Reply:
I guess I don't understand why we need both `AffinityCPUS` and `AffinityPreferredNumaNodes`. If these are mapping to the same setting under the hood, can we just expose the CPUs option, and let kubelet determine the node->cpu mapping? If these are mapping to different settings, what do they do differently such that we need both?

Reply:
This would be possible.  Due to the memory managers design today, to do this with no additional changes I believe it would require an additional query to remap the numa nodes that are calculated to the cpu's required.  

What do you see a major downside to including `AffinityPreferredNumaNodes`?

Reply:
I don't see a downside specifically to including `AffinityPreferredNumaNodes`, but I think if we are including it, we should understand what it really does and how it interacts with `AffinityCPUs`.

Reply:
I looked at it again, we could provide a way to map what the memory manager returns (which is a list of numanodes) to the CPU that numa node requires without any changes to memory manager. 

We can either make the system call or potentially even cache the CPU/numanode layout so we don't make the call for each container start (where numa nodes are required)

I don't really have a strong opinion so it comes down to where we want that conversion coming from. After some consideration, I am thinking that it may be more appropriate to do it in kubelet to make this interface simpler and easier to understand.

Reply:
We got the prototype working this week, After trying out a few difference scenarios, I am in favor of dropping this field.  I've updated the memory manager section in the KEP updates to identity how different scenarios would work: https://github.com/kubernetes/enhancements/pull/4738 PTAL